packages/amplify-category-auth/src/index.js

```diff
@@ -172,19 +160,19 @@ async function externalAuthEnable(context, externalCategory, resourceName, requi
     authProps = await removeDeprecatedProps(authProps);
     // replace secret keys from cli inputs to be stored in deployment secrets
 
-    let sharedParams = Object.assign({}, authProps);
+    let sharedParams = { ...authProps };
     privateKeys.forEach(p => delete sharedParams[p]);
     sharedParams = removeDeprecatedProps(sharedParams);
     // extracting env-specific params from parameters object
-    let envSpecificParams = {};
+    const envSpecificParams = {};
     const cliInputs = { ...sharedParams };
     ENV_SPECIFIC_PARAMS.forEach(paramName => {
       if (paramName in authProps) {
         envSpecificParams[paramName] = cliInputs[paramName];
         delete cliInputs[paramName];
       }
     });
-    context.amplify.saveEnvResourceParameters(context, category, authExists, envSpecificParams);
+    context.amplify.saveEnvResourceParameters(context, category, currentAuthName, envSpecificParams);
```

Comment:
Can we remove this call to replace with EnvironmentParameterManager for each category ?

Reply:
`saveEnvResourceParameters` has been migrated to use the param manager internally. It also does some additional stuff with the deployment secrets so just to avoid potentially messing up some flow there, I'd like to leave this as-is for now. I agree though that we should work to remove some of these other methods that abstract storing / retrieving env params